whisper/whisperv6/whisper.go

```diff
@@ -528,7 +560,22 @@ func (wh *Whisper) runMessageLoop(p *Peer, rw p2p.MsgReadWriter) error {
 			if cached {
 				p.mark(&envelope)
 			}
-		case p2pCode:
+		case powRequirementCode:
+			s := rlp.NewStream(packet.Payload, uint64(packet.Size))
+			i, err := s.Uint()
+			if err != nil {
+				log.Warn("failed to decode powRequirementCode message, peer will be disconnected", "peer", p.peer.ID(), "err", err)
+				return errors.New("invalid powRequirementCode message")
+			}
+			f := math.Float64frombits(i)
+			if math.IsInf(f, 0) || math.IsNaN(f) {
```

Comment:
And negative/zero values?

Reply:
zero allowed, but not negative

Reply:
My bad, I misread what `IsInf` was doing.

Reply:
Actually, no I didn't. You also need to make sure that the PoW is positive